src/app/shared/wysiwyg/prosemirror.markdown.editor.ts

```diff
@@ -246,153 +331,166 @@ export class ProseMirrorMarkdownEditor {
-      bind('Enter', splitListItem(schema.nodes.list_item));
-      bind('Mod-[', liftListItem(schema.nodes.list_item));
-      bind('Mod-]', sinkListItem(schema.nodes.list_item));
+    if (this.outputSchema.nodes.list_item) {
+      bind('Enter', splitListItem(this.outputSchema.nodes.list_item));
+      bind('Mod-[', liftListItem(this.outputSchema.nodes.list_item));
+      bind('Mod-]', sinkListItem(this.outputSchema.nodes.list_item));
     }
-    if (schema.nodes.paragraph) {
-      bind('Shift-Ctrl-0', setBlockType(schema.nodes.paragraph));
+    if (this.outputSchema.nodes.paragraph) {
+      bind('Shift-Ctrl-0', setBlockType(this.outputSchema.nodes.paragraph));
     }
-    if (schema.nodes.code_block) {
-      bind('Shift-Ctrl-\\', setBlockType(schema.nodes.code_block));
+    if (this.outputSchema.nodes.code_block) {
+      bind('Shift-Ctrl-\\', setBlockType(this.outputSchema.nodes.code_block));
     }
-    if (schema.nodes.heading) {
+    if (this.outputSchema.nodes.heading) {
       for (let i = 1; i <= 6; i++) {
-        bind('Shift-Ctrl-' + i, setBlockType(schema.nodes.heading, {level: i}));
+        bind('Shift-Ctrl-' + i, setBlockType(this.outputSchema.nodes.heading, {level: i}));
       }
     }
-    if (schema.nodes.horizontal_rule) {
-      bind('Mod-_', (state, onAction) => {
-        onAction(state.tr.replaceSelectionWith(schema.nodes.horizontal_rule.create()).scrollAction());
+    if (this.outputSchema.nodes.horizontal_rule) {
+      bind('Mod-_', (state, dispatchTransaction) => {
+        dispatchTransaction(state.tr.replaceSelectionWith(this.outputSchema.nodes.horizontal_rule.create()).scrollIntoView());
         return true;
       });
     }
 
     return keys;
   }
 
-  buildInputRules(schema) {
+  buildInputRules() {
     let result = [];
-    if (schema.nodes.blockquote) {
-      result.push(blockQuoteRule(schema.nodes.blockquote));
+    if (this.outputSchema.nodes.blockquote) {
+      result.push(blockQuoteRule(this.outputSchema.nodes.blockquote));
     }
-    if (schema.nodes.ordered_list) {
-      result.push(orderedListRule(schema.nodes.ordered_list));
+    if (this.outputSchema.nodes.ordered_list) {
+      result.push(orderedListRule(this.outputSchema.nodes.ordered_list));
     }
-    if (schema.nodes.bullet_list) {
-      result.push(bulletListRule(schema.nodes.bullet_list));
+    if (this.outputSchema.nodes.bullet_list) {
+      result.push(bulletListRule(this.outputSchema.nodes.bullet_list));
     }
-    if (schema.nodes.code_block) {
-      result.push(codeBlockRule(schema.nodes.code_block));
+    if (this.outputSchema.nodes.code_block) {
+      result.push(codeBlockRule(this.outputSchema.nodes.code_block));
     }
-    if (schema.nodes.heading) {
-      result.push(headingRule(schema.nodes.heading, 6));
+    if (this.outputSchema.nodes.heading) {
+      result.push(headingRule(this.outputSchema.nodes.heading, 6));
     }
     return result;
   }
 
   buildMenuItemsPlugin() {
-    let r = {};
+    let r = {}, deps;
```

Comment:
maybe slightly more descriptive variable names (is r for render and deps for dependencies?) ?